core/genesis.go

```diff
@@ -255,11 +263,14 @@ func (g *Genesis) configOrDefault(ghash common.Hash) *params.ChainConfig {
 
 // ToBlock creates the genesis block and writes state of a genesis specification
 // to the given database (or discards it if nil).
-func (g *Genesis) ToBlock(db ethdb.Database) *types.Block {
+func (g *Genesis) ToBlock(db ethdb.Database) (*types.Block, error) {
 	if db == nil {
 		db = rawdb.NewMemoryDatabase()
 	}
-	statedb, _ := state.New(common.Hash{}, state.NewDatabase(db), nil)
+	statedb, err := state.New(common.Hash{}, state.NewDatabase(db), nil)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I don't get this. How can this possibly error?

Reply:
if the root isn't in the database, the underlying `trie.OpenTrie` will return an error, and if this error isn't catch, you end up with `statedb = nil`, which will crash later. I spent too much time looking for where the problem was, it would have been immediately obvious had an error been returned here.

Reply:
But the root is `common.Hash{}`, right? So I still don't get it

Reply:
Right, and in `master`, an empty hash will return an empty trie. I got a problem when I modified `OpenTrie` to support Verkle trees, which don't have the same behavior (it's expecting a translated layer, not an empty tree).

I can certainly do the same thing in Verkle trees: return an empty tree when receiving `common.Hash`. This PR still makes sense, though, because if any other new database system appears, the same problem will occur.